app/controllers/workflow_executions_controller.rb

```diff
@@ -20,6 +20,8 @@ def index
   def show
     return unless @tab == 'files'
 
+    authorize! @workflow_execution, to: :read?
```

Comment:
**nitpick (blocking):** Can you move this above the return statement as we don't even want to show the default tab if they don't have permission